src/matrix_comps/realjordanform.jl

```diff
@@ -0,0 +1,59 @@
+"""
+    realjordanform(D) -> J
+
+Returns the real Jordan canonical form of the eigenvalues specified in
+the vector D.
+
+# Examples
+```julia
+julia> D = [-1, -1+2im, -1-2im, -1+2im, -1-2im];
+julia> realjordanform(D)
+5x5 Array{Int64,2}:
+ -1   0   0   0   0
+  0  -1  -2   1   0
+  0   2  -1   0   1
+  0   0   0  -1  -2
+  0   0   0   2  -1
+```
+"""
+
+function realjordanform{T<:Number}(D::AbstractVector{T})
+  sort!(D, by=imag)
+  sort!(D, by=x->abs(imag(x))) # make sure conjugated pairs are next to eachother
+  sort!(D, by=real)
+
+  J = zeros(real(T), length(D), length(D))
```

Comment:
3 calls to `length(D)`. most likely the compiler optimizes them out; but just wanted to say. nothing major :)